include/tokenizer.h

```diff
@@ -44,27 +42,33 @@ class Tokenizer
             EOL
         };
 
-        struct KeywordToken
-        {
+        struct KeywordToken {
                 std::string keyword;
                 TokenType token;
         };
 
         using TokenData = std::variant<std::monostate, std::string, int, char>;
 
-        // Token data access
-        int variable_num() const;
-        const std::string& get_string() const;
-        int get_num() const;
-        const TokenData& get_token_data() const;
-        std::string token_to_string(TokenType token) const;
+        // Token operations
+        TokenType current_token() const { return current_token_; }
```

Comment:
In our backlog i have that issue for performance, using `constexpr` as much as possible especially for functions like these help a ton. 

But maybe in a side project, once version 1.0 is done use a performance benchmark tool to actually test if you can make the interpreter faster, that'll give you some real world experience.